azurerm/resource_arm_managed_cluster.go

```diff
@@ -0,0 +1,556 @@
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmManagedClusterCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	managedClusterClient := client.managedClustersClient
+
+	log.Printf("[INFO] preparing arguments for Azure ARM AKS managed cluster creation.")
+
+	resGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+	dnsPrefix := d.Get("dns_prefix").(string)
+	kubernetesVersion := d.Get("kubernetes_version").(string)
+
+	linuxProfile := expandAzureRmManagedClusterLinuxProfile(d)
+	agentProfiles := expandAzureRmManagedClusterAgentProfiles(d)
+
+	tags := d.Get("tags").(map[string]interface{})
+
+	parameters := containerservice.ManagedCluster{
+		Name:     &name,
+		Location: &location,
+		ManagedClusterProperties: &containerservice.ManagedClusterProperties{
+			DNSPrefix:         &dnsPrefix,
+			KubernetesVersion: &kubernetesVersion,
+			LinuxProfile:      &linuxProfile,
+			AgentPoolProfiles: &agentProfiles,
+		},
+		Tags: expandTags(tags),
+	}
+
+	servicePrincipalProfile := expandAzureRmManagedClusterServicePrincipal(d)
+	if servicePrincipalProfile != nil {
+		parameters.ServicePrincipalProfile = servicePrincipalProfile
+	}
+
+	ctx := client.StopContext
+	_, error := managedClusterClient.CreateOrUpdate(ctx, resGroup, name, parameters)
+	if error != nil {
+		return error
+	}
+
+	read, err := managedClusterClient.Get(ctx, resGroup, name)
+	if err != nil {
+		return err
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read AKS managed cluster %s (resource group %s) ID", name, resGroup)
+	}
+
+	log.Printf("[DEBUG] Waiting for AKS managed cluster (%s) to become available", d.Get("name"))
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"Updating", "Creating"},
+		Target:     []string{"Succeeded"},
+		Refresh:    ManagedClusterStateRefreshFunc(client, resGroup, name),
+		Timeout:    30 * time.Minute,
+		MinTimeout: 15 * time.Second,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for AKS managed cluster (%s) to become available: %s", d.Get("name"), err)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmManagedClusterRead(d, meta)
+}
+
+func resourceArmManagedClusterRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	managedClusterClient := meta.(*ArmClient).managedClustersClient
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["managedClusters"]
+
+	ctx := client.StopContext
+	resp, err := managedClusterClient.Get(ctx, resGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on Azure Container Service %s: %s", name, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("location", azureRMNormalizeLocation(*resp.Location))
```

Comment:
`Location` isn't always returned from the API (even though the API spec says it's guaranteed) - can we make this:

```
if location := resp.Location; location != nil {
  d.Set("location", *location)
}
```

Reply:
Thanks, fixed.